designs/containers/vnf-deployment/vnf-deployment.md

```diff
@@ -0,0 +1,106 @@
+
+### Nuage SDN for Kubernetes
+Nuage does not present a restriction with respect to where the container VNF should be deployed. 
+
+
+## Design Changes
+The implementation of this feature will require changes on the OSC entities and operations involved on the definition and registration of VNF images; deployment specification; and monitoring of container VNFs. 
+
+### Container VNF Images
+In order for a user to deploy a VM VNF using OSC they must register the image file of the VNF along with some additional metadata. For container VNFs this model will be further simplified.  The image for the container is not necessary as it will be retrieved by the virtualization platform from a Docker registry, the user will simply need to provide the metadata needed for OSC to match the VNF with the supported security manager and virtualization platform versions and create the [K8s Deployment object](#kubernetes-deployment).
+This new model is implemented by a couple of new **POST** REST APIs to create the existing types `ApplianceDTO` and `ApplianceSoftwareVersionDTO`. The choice of using two APIs instead of a single one to create both objects is consistent with the existing APIs to GET and DELETE these resources, this also allows the clients to manage appliances and their software versions independently.  
+
+### REST API  
+
+#### Appliance
+The resource `/api/server/v1/catalog/{applianceId}/` will have a create/POST API implemented by the class `ApplianceApis`.  This API implementation will make use of the new service `AddApplianceService`.  
+
+
+#### Appliance Software Version
+The resource `/api/server/v1/catalog/{applianceId}/versions/{ApplianceSoftwareVersionId}` will have a create/POST API implemented by the class `ApplianceApis`.  This API implementation will make use of the new service `AddApplianceSoftwareVersionService`.  
+In addition to that, the object `ApplianceSoftwareVersionDto` will include a new field: `imagePullSecretName`. This optional field can be used for clients to provide the name of the secret to be used by Kubernetes to [retrieve images from a private Docker registry](#private-docker-registry). 
+
+
+```java
+ApplianceSoftwareVersionDto {
+id (integer, optional),
+parentId (integer, optional),
+swVersion (string),
+virtualizationType (string) = 
+['OPENSTACK', 'KUBERNETES'],
+virtualizationVersion (string),
+imageUrl (string),                            // For container VNFs this should be the Docker image name TODO: ref
+imagePullSecretName (string, optional),       // New field added for container VNFS. Contains the name of the key used by the virtualization platform to pull the container images from a private registry.
+encapsulationTypes (Array[string], optional),
+imageProperties (object, optional),

```

Comment:
There may be scenarios in the future where tags/labels can be added to deployed VNFS. Depending on the scenario that might be defined by the image or on the deployment spec.  I have not identified yet anything that needs to provided during the bootstrap of the images, if that becomes needed we can use this same field (bootstrap properties) hopefully.